homeassistant/components/dht/sensor.py

```diff
@@ -160,18 +161,27 @@ def update(self):
 class DHTClient:
     """Get the latest data from the DHT sensor."""
 
-    def __init__(self, adafruit_dht, sensor, pin):
+    def __init__(self, sensor, pin):
         """Initialize the sensor."""
-        self.adafruit_dht = adafruit_dht
         self.sensor = sensor
-        self.pin = pin
+        self.pin = getattr(board, pin)
         self.data = {}
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data the DHT sensor."""
-        humidity, temperature = self.adafruit_dht.read_retry(self.sensor, self.pin)
-        if temperature:
-            self.data[SENSOR_TEMPERATURE] = temperature
-        if humidity:
-            self.data[SENSOR_HUMIDITY] = humidity
+        dht = self.sensor(self.pin)
+
+        try:
+            humidity = dht.humidity
+            temperature = dht.temperature
+            if temperature:
+                self.data[SENSOR_TEMPERATURE] = temperature
+            if humidity:
+                self.data[SENSOR_HUMIDITY] = humidity
+        except RuntimeError as error:
+            pass
```

Comment:
Don't pass. At minimum we need to log something here. Why do we catch RuntimeError?

Reply:
Their docs say that pulling from the dht devices is often buggy / throws errors that can be ignored and their code example shows catching RuntimeError and retrying. I am removing it because given the type of sensor it doesn't seem critical to retry it. The original library had a retry option which just slept for 2 seconds and retried again so I could implement that as well, but I am assuming it isn't necessary.

Reply:
Ok, at minimum we should log the error at debug level. We already do that below, so then we can remove this catch.